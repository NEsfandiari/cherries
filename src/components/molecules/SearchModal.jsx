import React, { Component } from 'react'
import styled from 'styled-components'
import { IoIosClose } from 'react-icons/io'

const Container = styled.div`
  width: 100%;
  padding: 1rem;
  .popup {
    z-index: 10;
    position: fixed;
    width: 100%;
    height: 300px;
    background-color: #fbe5e9;
  }
  .popupInner {
    margin: 0 auto;
    display: flex;
    justify-content: center;
    align-items: center;
    align-content: center;
    flex-direction: column;
    width: 75%;
  }

  input {
    text-align: center;

    background-color: #fbe5e9;
    border: 0px;
    border-bottom: 2px solid black;
    width: 100%;
  }

  #search-form {
    width: 100%;
    position: relative;
    font-size: 4rem;
  }
  input:focus {
    outline: none;
    border-bottom: 4px solid black;
  }
  #search-image {
    width: 28px;
    height: 28px;
  }

  #search-button {
    background-color: transparent;
    border: 0px;
    margin: 0 auto;
  }

  #search-button:hover {
    cursor: pointer;
  }
  #close-button:hover {
    cursor: pointer;
  }
  #button-container {
    text-align: center;
    width: 100%;
  }
  #close-button {
    height: 50px;
    width: 50px;
    margin-right: 2rem;
  }
  #search-bar-text {
    text-align: center;
    font-weight: bold;
    padding-top: 1rem;
    line-height: 0.1rem;
  }

  .grayed-out {
    animation: fadein 0.2s;

    @keyframes fadein {
      from {
        opacity: 0;
      }
      to {
        opacity: 1;
      }
    }
  }
`

class SearchModal extends Component {
  constructor(props) {
    super(props)
    this.state = {
      search: '',
    }
  }

  handleChange = evt => {
    this.setState({ [evt.target.name]: evt.target.value })
  }

  //handleClick will call togglePopup which will change state of parent component NavButtons
  handleClick = () => {
    this.props.togglePopup()
  }

  render() {
    const style = {
      display: 'flex',
      position: 'fixed',
      top: 0,
      bottom: 0,
      left: 0,
      right: 0,
      width: '100vw',
      height: '100vh',
      backgroundColor: 'rgba(255, 255, 255, 0.9)',
      zIndex: 100000,
    }
    return (
      <Container>
        <div className="grayed-out" style={style}>
          <div className="popup">
            <div>
              <IoIosClose id="close-button" onClick={this.handleClick} />
            </div>
            <div className="popupInner">
              <form id="search-form" action="/catalog">
                <div id="button-container">
                  <button id="search-button">
                    <img id="search-image" src={this.props.searchIcon} alt="" />
                  </button>
                </div>
                <input
                  onChange={this.handleChange}
                  name="search"
                  type="text"
                  autofocus="autofocus"
                />
              </form>
              <div id="search-bar-text">
                <p>What are you looking for?</p>
                <p>Product name, type, color (or anything)</p>
              </div>
            </div>
          </div>
        </div>
      </Container>
    )
  }
}

export default SearchModal
