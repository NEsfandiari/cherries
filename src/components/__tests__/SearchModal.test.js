import React from 'react'
import renderer from 'react-test-renderer'
import SearchModal from '../molecules/SearchModal'

describe('SearchModal', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<SearchModal />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
