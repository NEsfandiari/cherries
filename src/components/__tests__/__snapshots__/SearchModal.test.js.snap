// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchModal renders correctly 1`] = `
<div
  className="sc-bdVaJa eOdnK"
>
  <div
    className="grayed-out"
    style={
      Object {
        "backgroundColor": "rgba(255, 255, 255, 0.9)",
        "bottom": 0,
        "display": "flex",
        "height": "100vh",
        "left": 0,
        "position": "fixed",
        "right": 0,
        "top": 0,
        "width": "100vw",
        "zIndex": 100000,
      }
    }
  >
    <div
      className="popup"
    >
      <div>
        <svg
          fill="currentColor"
          height="1em"
          id="close-button"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
        >
          <path
            d="M278.6 256l68.2-68.2c6.2-6.2 6.2-16.4 0-22.6-6.2-6.2-16.4-6.2-22.6 0L256 233.4l-68.2-68.2c-6.2-6.2-16.4-6.2-22.6 0-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3l68.2 68.2-68.2 68.2c-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3 6.2 6.2 16.4 6.2 22.6 0l68.2-68.2 68.2 68.2c6.2 6.2 16.4 6.2 22.6 0 6.2-6.2 6.2-16.4 0-22.6L278.6 256z"
          />
        </svg>
      </div>
      <div
        className="popupInner"
      >
        <form
          action="/catalog"
          id="search-form"
        >
          <div
            id="button-container"
          >
            <button
              id="search-button"
            >
              <img
                alt=""
                id="search-image"
              />
            </button>
          </div>
          <input
            autofocus="autofocus"
            name="search"
            onChange={[Function]}
            type="text"
          />
        </form>
        <div
          id="search-bar-text"
        >
          <p>
            What are you looking for?
          </p>
          <p>
            Product name, type, color (or anything)
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
